agent/xds/testdata/listeners/custom-upstream.envoy-1-18-x.golden

```diff
@@ -41,14 +41,13 @@
               "name": "envoy.filters.network.tcp_proxy",
               "typedConfig": {
                 "@type": "type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy",
-                "statPrefix": "upstream.prepared_query_geo-cache",
-                "cluster": "geo-cache.default.dc1.query.11111111-2222-3333-4444-555555555555.consul"
+                "statPrefix": "foo-stats",
+                "cluster": "random-cluster"
```

Comment:
Where did this change come from? 🤔

Reply:
Previously the test setup was using escape hatch for only the first upstream:
```
			snap.Proxy.Upstreams[0].Config["envoy_listener_json"] =
				customListenerJSON(t, customListenerJSONOptions{
					Name: "custom-upstream",
				})
```
So the 2nd upstream (Prepared Query) was never being tested with the escape hatch.
I changed the test to apply the same escape hatch json to all the upstreams

Reply:
Ah, got it!